python/ray/tune/function_runner.py

```diff
@@ -85,97 +76,48 @@ def run(self):
             self._status_reporter._done = True
 
 
-def import_function(file_path, function_name):
-    # strong assumption here that we're in a new process
-    file_path = os.path.expanduser(file_path)
-    sys.path.insert(0, os.path.dirname(file_path))
-    if hasattr(importlib, "util"):
-        # Python 3.4+
-        spec = importlib.util.spec_from_file_location(
-            "external_file", file_path)
-        external_file = importlib.util.module_from_spec(spec)
-        spec.loader.exec_module(external_file)
-    elif hasattr(importlib, "machinery"):
-        # Python 3.3
-        from importlib.machinery import SourceFileLoader
-        external_file = SourceFileLoader(
-            "external_file", file_path).load_module()
-    else:
-        # Python 2.x
-        import imp
-        external_file = imp.load_source("external_file", file_path)
-    if not external_file:
-        raise TuneError("Unable to import file at {}".format(file_path))
-    return getattr(external_file, function_name)
-
-
-class ScriptRunner(Agent):
-    """Agent that runs a user script returning training results."""
-
-    _agent_name = "script"
+class FunctionRunner(Trainable):
+    """Trainable that runs a user function returning training results.
```

Comment:
Would be nice to document the following:
1.  the fact that functions are run on a separate thread here
2. some form of example usage
3. why save/restore are hard to implement

Reply:
This is an internal class, so this is not necessary.